treeopt/treeOptExample.py

```diff
@@ -0,0 +1,25 @@
+from treeOpt import TreeOpt
```

Comment:
Please remove this from `treeopt`, and do one (or both) of the following:

 - Add a directory called `examples` in the root of the repo and move this file there.
 - Add it to the sphinx documentation

Note that if you do both, you need to update the code in two places. There are some hacky ways to include python files or jupyter notebooks from an example directory directly in the documentation, but I don't think we need to worry about it right now. 

BTW: This is a really cool Sphinx-theme, that lets you use Markdown and Jupyter Notebooks instead of reStructured Text: https://github.com/executablebooks/sphinx-book-theme I really like the look of it and you can execute code examples right in the documentation.

Reply:
I had some problems with the relative imports when mooving the file. How do I tell python to import a file from a its parent directory?

Reply:
We shouldn't use relative imports at all. All imports should be relative to the main module, even imports within the module. So the import in the example should look like this:

`from treeopt.treeOpt import TreeOpt`

This is a bit ugly, since you have to type treeopt three times, each time with a different capitalization.

Since the `TreeOpt` class is so central to your code, you could import it directly into your main module by adding the following lines to your `treeopt.__init__.py`:

```
from treeopt.treeOpt import *  # noqa:  F401
__all__ = ["TreeOpt"]  # noqa F405
```

With these changes, `from treeopt import TreeOpt` suffices.

---

That said, we should change two more things to your imports and module structure:

1. Always import from the main module, e.g. replace lines like this:

    https://github.com/hbrs-cse/treeopt/blob/2ecca2be43e6f9acee08c7ba95bd34ea4e512029/treeopt/treeOpt.py#L9

    by `import treeopt.modules.sampling as sampling`.
2. Try to get rid of the `modules` subdirectory. All python files within the `treeopt` directory are already submodules, and having to type `modules` adds an extra step and provides no more structure to the code.

Reply:
I still think we should add the example to a new `examples` directory. It is not really a unit test and users of the library will be looking for examples. They might not find it in a directory called `tests`